File: oti/tp1_qunit/money_js/src/money.js
var money = function (v, curr) {
    this.v = v;
    this.curr = curr;
}

money.prototype.getCurrency =
    function () {
        return this.curr;
    };

money.prototype.getValue =
    function () {
        return this.v;
    };

/**
 * Add currency comparison
 * @param otherM
 * @returns {boolean}
 */
money.prototype.equals =
    function (otherM) {
        if (typeof(otherM) == "undefined") {
            return false;
        }
        return (otherM.getValue() == this.getValue() &&
            otherM.getCurrency().toLowerCase() === this.getCurrency().toLowerCase());
    };

money.prototype.toString =
    function () {
        return this.getValue() + " (" + this.getCurrency() + ")";
    };

File: oti/tp1_money_java/src/main/java/fr/ulille1/fil/odeva/Money.java
package fr.ulille1.fil.odeva;

public class Money {
    private int value;
    private String currency;


    Money(int value, String currency) {
        this.value = value;
        this.currency = currency;
    }

    public int getValue() {
        return this.value;
    }

    public String getCurrency() {
        return this.currency;
    }

    public String toString() {
        return this.getValue() + " (" + this.getCurrency() + ")";
    }

    /**
     * modify the default equals
     * @param obj
     * @return
     */
    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof Money))
            return false;
        if (obj == this)
            return true;

        Money m = (Money) obj;
        return m.getValue() == this.getValue() && m.getCurrency() == this.getCurrency();
    }

}
